### ISSUE TITLE
`win_compat_test.cpp` generates warning since `Windows.h` brings in conflicting `max()` macro

### ISSUE THREAD

**Issue by helmesjo** [boost-ext/ut#579]
## Expected Behavior
It compiles fine.

## Actual Behavior
```
boost/ut.hpp(708): warning C4003: not enough arguments for function-like macro invocation 'max'
```

## Steps to Reproduce the Problem

  1. Build `master` (including tests) with MSVC.

## Specifications

  - Version: master (commit [92844cb90f](https://github.com/boost-ext/ut/commit/bf8388f61103571dee3061a4ef23292a320d9dbf)).
  - Platform: Windows

**Comment by helmesjo** [comment]
@kris-jusiak Any possibility to get this fixed and in to a `v2.0.1` release? Right now, if building with `-WX` (warning as error), it gets blocked by this warning.

**Comment by krzysztof-jusiak** [comment]
Sure @helmesjo, I don't have neither windows nor msvc access, could you please create a PR so we can be sure it's fixed? No issues with releasing 2.0.1 when the fix is in, thanks

**Comment by helmesjo** [comment]
Created PR: #601
